carapace-server/src/main/java/org/carapaceproxy/client/impl/EndpointConnectionImpl.java

```diff
@@ -235,7 +236,6 @@ public void sendRequest(HttpRequest request, RequestHandler clientSidePeerHandle
                     RequestHandler _clientSidePeerHandler = clientSidePeerHandler;
                     if (!future.isSuccess()) {
                         LOG.log(Level.INFO, this + " sendRequest " + request.getClass() + " failed", future.cause());
-                        changeStateTo(ConnectionState.RELEASABLE, ConnectionState.REQUEST_SENT);
```

Comment:
Whether the server callback runs first the state might be already set to DALAYED_RELEASE. So This is useless.
Whether the server callback runs last it will perform a release on a yet closed channel, throwing an error.